### PR TITLE
Began generic function filter.R to filter on multiple attributes (up …

### PR DESCRIPTION
**Pull request by grrmartin-USGS**
…to 5).  Crude approach will apply filter command within if-clauses in function statement, but gives error when same filter command is called when executing the filter function as coded here. Get: Error in UseMethod("filter_") :

#' # no applicable method for 'filter_' applied to an object of class "character". Tidyverse does not allow input of attributes  as variables to filter using the .data format.  Also attempted to write loop to process filter command on multiple attributes, but could not implement in .data format of tidyverse.

### CODE REVIEW

File: R/filter.R [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+#'  #.data[[attributes[[3]]]] == conditions[[3]]
+#'  #)
+#'  
+#'  
+#'
+#'  
+#'  
+#'  
+#'  
+filtered <- function(s.wuds, attributes, conditions){
```
Comment (ldecicco-USGS) [inline R/filter.R:51]: I try to avoid using the `%>%` in functions. It makes troubleshooting easier for the user later.

File: R/filter.R [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+  )
+  }
+  
+  if (length(attributes)==3){s.wuds %>% filter(
+    .data[[attributes[[1]]]] == conditions[[1]],
+    .data[[attributes[[2]]]] == conditions[[2]],
+    .data[[attributes[[3]]]] == conditions[[3]]
+  )
+  } 
+  if (length(attributes)==4){s.wuds %>% filter(
```
Comment (ldecicco-USGS) [inline R/filter.R:70]: Honestly, I'm not a big fan of any of these filter functions. My impression is that for long-term maintenance/readability/etc....using the basic dplyr filter function is safest. 

For example, I find this less confusing:
```r
s_wuds_filtered <- s.wuds %>%
               filter(FROM_NAT_WATER_USE_CD == "WS",
                      FROM_COUNTY_NM == "Lawrence County",
                      YEAR == "1990")
```
Than what this swuds_filter would look like:
```r
s_wuds_filtered <- s.wuds %>%
                   filtered(c("FROM_NAT_WATER_USE_CD", "FROM_COUNTY_NM", "YEAR"),
                            c("WS", "Lawrence County", "1990"))

```

**Comment by ldecicco-USGS** [comment]
Here's my vote. Let's keep the monthly timeseries function, and lose ALL filtering functions. Now that we have a "swuds" object, I don't see how either the generic one or the individual ones make our lives easier than just using the vanilla "dplyr" filter.

**Comment by ldecicco-USGS** [comment]
So here's what I'm going to do, I'll merge this PR...delete all of the filter functions, and spruce up the monthly timeseries.  Sound OK?

**Comment by grrmartin-USGS** [comment]
OK. Can the vanilla "dplyr" filter be implemented within a loop over multiple attributes-conditions? The SWUDS data have a certain hierarchy that user's can navigate through to explore it.  For example, the basic categories of water-use are defined in FROM_NAT_WATER_USE_CD,(c39 nwis) including WS (water supply[public]), IN (industrial), IR (irrigation), etc.  The Ohio sample data have multiple categories of water use (c39)--we are focusing initially on public supply (WS), which is most of the data. Also, Ohio sample data are for surface-water intakes only (the full dataset with wells is much larger)--so site-type codes (FROM_SITE_TP_CD) is also an attribute the user will want to filter to drill down into the data. Major site-type codes include FA-DV (sw intakes) and GW (wells). [One function I'm going to make will do an inventory of the state's SWUDS data listing types and counts of sites, types and counts of conveyances from and to specific site-types. We we will want a tabular display in shiny.]  

Ideally, the user will be able to filter and aggregate the data in various sequences to explore and look for anomalies. For example, a user may want to identify the PS intakes in a specific county, then seeing that group listed/mapped, select a particular intake by permit number or site number or station name, to graph the water use. Alternatively, because water use for individual wells/intakes may be highly varied over time, the user may prefer to graph water use aggregated by permit, or county, first, to look for outliers--then if they see a county with odd values, filter to look at individual sites in that county.  Is this mode of operation---filtering and aggregating in various sequences, and revising the sequence later in the session---something we can do with R/shiny?   

Part of the motivation to make this tool is that $5 million has been spent to pay WSC to build SWUDS datasets for PS for year 2010, and the data has not been reviewed.  The local Water-Use Specialist may not have a grasp of their whole SWUDS content, partly because the data are loaded in multiple pieces and funding for the work was spread over multiple years.

**Comment by ldecicco-USGS** [comment]
Yes, absolutely.

**Comment by ldecicco-USGS** [comment]
If you make specific examples of what you want to see from the OH data set, I can write you the code you need.

**Comment by grrmartin-USGS** [comment]
OK great!  We had a working list of 'User Stories' that included requests for functionality from regional, local Water-Use Specialists.  I'll check with Jen on status of list and how to best represent what is needed--maybe we should prioritize the functionality requests.

**Comment by jshourds-usgs** [comment]
This [link](https://docs.google.com/document/d/13vAOSP-mXV7OSiWSXBgur-ZtDG6PcS6TKh3HtOYDPlQ/edit) should be open to everyone.  It's a doc of the user stories we came up with during the Sprint here in May.

**Pull request merged**